src/porepy/examples/terzaghi_biot.py

```diff
@@ -542,8 +539,10 @@ def bc_type_mechanics(self, sd: pp.Grid) -> pp.BoundaryConditionVectorial:
             the South, and rollers on the sides.
 
         """
-        # Inherit bc from parent class. This sets all bc faces as Dirichlet.
-        bc = super().bc_type_mechanics(sd=sd)
+        # Start with all faces as Dirichlet faces, analogous to base mechanics set-up.
+        boundary_faces = self.domain_boundary_sides(sd).all_bf
+        bc = pp.BoundaryConditionVectorial(sd, boundary_faces, "dir")
+        bc.internal_to_dirichlet(sd)
```

Comment:
Why do you prefer this?

Reply:
Eliminating the need to call super and the inheritance from `BoundaryConditionsMechanics`.

I also tried `BoundaryConditionsMechanics.bc_type_mechanics(self,..)` with `self` as an explicit argument, but mypy was giving me hell.

Then I wagered that 3 lines of duplicate code are worth resolving the loop in the inheritance.